xc/xenops_server_xen.ml

```diff
@@ -1211,10 +1212,14 @@ module VM = struct
       | HVM hvm_info ->
         let disks = List.filter_map (fun vbd ->
             let id = vbd.Vbd.id in
-            if hvm_info.Vm.qemu_disk_cmdline && (List.mem_assoc id qemu_vbds)
+            if (List.mem_assoc id qemu_vbds)
             then
-              let index, bd = List.assoc id qemu_vbds in
-              let path = block_device_of_vbd_frontend bd in
+              let index, bd_opt = List.assoc id qemu_vbds in
+              let path =
+                match bd_opt with
+                | Some bd -> block_device_of_vbd_frontend bd
+                | None -> ""
+              in
```

Comment:
it would be useful to comment in the commit message the gist of what this patch is doing. For what I see:
- when plugging the vbds, we are now mapping all the vbds to VM.non_persistent.qemu_vbds, even the vbds that are an empty cdrom (which were never added to qemu_vbds before this patch). The result is that we now always have the index for the empty cdroms, that we need to use to build the qemu command line.
- when unplugging the vbds, we remove the vbd from VM.non_persistent.qemu_vbds, but are not destroying the frontend of an empty cdrom.

Questions:
- the frontend that is not destroyed during unplug is destroyed when the cdrom is ejected?
- during a cdrom mount/eject operation, do we need to change the corresponding vbd entry in VM.non_persistent.qemu_vbds to keep its path synchronised?

Reply:
I tried the following tests on a VM with one hard drive:
* start the VM with empty cdrom, then mount a disk, then shutdown
* start the VM with full cdrom, then eject the disk, then shutdown
In the first case 1 qemu frontend is created and 1 is destroyed, in the second case 2 are created and 2 are destroyed, so on the surface of it, it seems the mount/eject operations while the VM is running does not create/destroy a frontend.

Reply:
if the behaviour of create_frontend and destroy_frontend is the same as before, then the patch is ok.